pkg/operator/ceph/object/rgw-probe.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+
+PROBE_TYPE="{{ .ProbeType }}"
+PROBE_PORT="{{ .Port }}"
+PROBE_PROTOCOL="{{ .Protocol }}"
+
+# standard bash codes start at 126 and progress upward. pick error codes from 125 downward for
+# script as to allow curl to output new error codes and still return a distinctive number.
+USAGE_ERR_CODE=125
+PROBE_ERR_CODE=124
+# curl errors: 1-123
+
+STARTUP_TYPE='startup'
+READINESS_TYPE='readiness'
+case "$PROBE_TYPE" in
+"$STARTUP_TYPE")
+  # For the startup probe, treat any error other than specifically okay responses as an error that
+  # should prevent starting the RGW. Being able to successfully query the RGW to get its status is
+  # critical for monitoring its health. If there is any issue that prevents the check from
+  # happening, it's good to flag that for the user early since it's unlikely health checking will
+  # work in probes during runtime.
+  true # noop: value ok
+  ;;
+"$READINESS_TYPE")
+  # For the readiness probes, the RGW should be removed from the load balancer (service) pool if
+  # clients can't reach it to eliminate unnecessary client I/O failures.
+  true # noop: value ok
+  ;;
+*)
+  echo "ERROR: invalid probe type $PROBE_TYPE" >/dev/stderr
+  exit $USAGE_ERR_CODE
+  ;;
+esac
+
+if [[ -z "$PROBE_PORT" ]]; then
+  echo "ERROR: port is unspecified" >/dev/stderr
+  exit $USAGE_ERR_CODE
+fi
+
+HTTP_TYPE="HTTP"
+HTTPS_TYPE="HTTPS"
+case "$PROBE_PROTOCOL" in
+"$HTTP_TYPE")
+  true # noop: value ok
+  ;;
+"$HTTPS_TYPE")
+  true # noop: value ok
+  ;;
+*)
+  echo "ERROR: invalid protocol type $PROBE_PROTOCOL" >/dev/stderr
+  exit $USAGE_ERR_CODE
+  ;;
+esac
+
+RGW_URL="$PROBE_PROTOCOL://0.0.0.0:$PROBE_PORT"
+
+function check() {
+  local URL="$1"
+  # --insecure - don't validate ssl if using secure port only
+  # --silent - don't output progress info
+  # --output /dev/stderr - output HTML header to stdout (good for debugging)
+  # --write-out '%{response_code}' - print the HTTP response code to stdout
+  curl --insecure --silent --output /dev/stderr --write-out '%{response_code}' "$URL"
+}
+
+http_response="$(check "$RGW_URL")"
+ec=$?
```

Comment:
Could we give it a longer name? To me, `ec` is erasure coding.
```suggestion
error_code=$?
```